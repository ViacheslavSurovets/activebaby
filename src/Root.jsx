import React, { Suspense, useMemo } from 'react';
import { useTranslation } from 'react-i18next';
import { memoizeWith, identity } from 'ramda';
import { Switch, Route, Redirect } from 'react-router-dom';
import styled from 'styled-components';
import { Header } from '@components';

const Layout = styled.div`
  flex: 1;
  display: flex;
  flex-direction: column;
`;

const MainContentWrapper = styled.main`
  height: 100%;
  margin-top: 3rem;
  margin-bottom: 3rem;
`;

const lazy = memoizeWith ( identity, ( path ) =>
  React.lazy ( () =>
    import(`@pages/${ path }`) ) );


const Root = () => {
  const { t, i18n } = useTranslation ();

  const headerCategory = useMemo ( () => ([
    {
      id: 'Прогулки',
      listItems: [
        {
          text: t ( 'header.headerCategory.columnOne.strollers' ),
          path: '#'
        },
        {
          text: t ( 'header.headerCategory.columnOne.quatro' ),
          path: '#'
        },
        {
          text: t ( 'header.headerCategory.columnOne.lorelli' ),
          path: '#'
        },
        {
          text: t ( 'header.headerCategory.columnOne.carello' ),
          path: '#'
        },
      ]
    },
    {
      id: 'Модульные',
      listItems: [
        {
          text: t ( 'header.headerCategory.columnTwo.modules' ),
          path: '#'
        },
        {
          text: t ( 'header.headerCategory.columnTwo.adamex' ),
          path: '#'
        },
        {
          text: t ( 'header.headerCategory.columnTwo.used' ),
          path: '#'
        }
      ]
    },
    {
      id: 'Автолюльки',
      listItems: [
        {
          text: t ( 'header.headerCategory.columnThree.carSeats' ),
          path: '#'
        },
        {
          text: t ( 'header.headerCategory.columnThree.carlo' ),
          path: '#'
        },
        {
          text: t ( 'header.headerCategory.columnThree.adamex' ),
          path: '#'
        },
        {
          text: t ( 'header.headerCategory.columnThree.used' ),
          path: '#'
        },
      ]
    },
    {
      id: 'Аксесуары',
      listItems: [
        {
          text: t ( 'header.headerCategory.columnFour.accessories' ),
          path: '#'
        },
        {
          text: t ( 'header.headerCategory.columnFour.adapters' ),
          path: '#'
        },
        {
          text: t ( 'header.headerCategory.columnFour.tires' ),
          path: '#'
        },
        {
          text: t ( 'header.headerCategory.columnFour.cameras' ),
          path: '#'
        }
      ],
    },
    {
      id: 'Постельное белье',
      listItems: [
        {
          text: t ( 'header.headerCategory.columnFive.linens' ),
          path: '#'
        },
        {
          text: t ( 'header.headerCategory.columnFive.individualTailoring' ),
          path: '#'
        }
      ]
    }
  ]), [ i18n.language ] );
  const headerArticles = useMemo ( () => ([
    {
      text: t ( 'header.headerArticle.articleOne.title' ),
      path: '#'
    },
    {
      text: t ( 'header.headerArticle.articleTwo.title' ),
      path: '#'
    },
    {
      text: t ( 'header.headerArticle.articleThree.title' ),
      path: '#'
    }

  ]), [ i18n.language ] );

  return (
    <Layout>
      <Header headerCategory={ headerCategory } headerArticles={ headerArticles } />
      <Suspense fallback='loading'>
      </Suspense>
      <React.Suspense fallback="Root content suspense">
        <MainContentWrapper>
          <Switch>
            <Route exact path="/">
              <Redirect to="/prams" />
            </Route>
            <Route path="/prams" component={ lazy ( 'Prams' ) } />
            <Route path="/map" component={ lazy ( 'Map' ) } />
          </Switch>
        </MainContentWrapper>
      </React.Suspense>
    </Layout>
  );
};

export default Root;
