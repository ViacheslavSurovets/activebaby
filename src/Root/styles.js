import styled from 'styled-components';

export const Layout = styled.div`
  flex: 1;
  display: flex;
  flex-direction: column;
  height: 100%;
  overflow: auto;
  overflow-x: hidden;
  position: relative;
  min-height: 50rem;
`;

export const MainContentWrapper = styled.main`
  display: flex;
  flex-direction: column;
  margin-top: 3rem;
  margin-bottom: 3rem;
`;
