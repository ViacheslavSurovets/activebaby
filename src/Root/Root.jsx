import React, { useRef, useState } from 'react';
import { memoizeWith, identity } from 'ramda';
import { Switch, Route, Redirect } from 'react-router-dom';

import { Header, OnScrollHeader, Footer } from '@components';
import { Layout, MainContentWrapper } from './styles';

const lazy = memoizeWith ( identity, ( path ) =>
  React.lazy ( () =>
    import(`@pages/${ path }`) ) );


const Root = () => {

  const [ menuTopVisibility, setMenuTopVisibility ] = useState ( false );

  const layout = useRef ( '' );
  const handleScroll = async () => {
    if ( layout.current.scrollTop >= 250 ) {
      await setMenuTopVisibility ( true );

    } else {
      await setMenuTopVisibility ( false );
    }
  };

  return (
    <Layout id='js-layout' ref={ layout } onScroll={ handleScroll }>
      <Header />

      <React.Suspense fallback="Root content suspense">
        <MainContentWrapper>

          <OnScrollHeader visible={ !!menuTopVisibility } />

          <Switch>
            <Route exact path="/">
              <Redirect to="/prams" />
            </Route>
            <Route path="/prams" component={ lazy ( 'Prams' ) } />
            <Route path="/map" component={ lazy ( 'Map' ) } />
          </Switch>
        </MainContentWrapper>
      </React.Suspense>
      <Footer />
    </Layout>
  );
};

export default Root;
