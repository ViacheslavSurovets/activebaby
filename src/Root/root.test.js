import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import Root from './Root';
import configureMockStore  from 'redux-mock-store';


const mockStore = configureMockStore();
const store = mockStore({});

describe('BurgerMenuComponent', ()=> {
  it('Burger Menu Snapshot', ()=> {
    expect ( shallow (
      <Provider store={store}>
        <Root  />
      </Provider>
    ) ).toMatchSnapshot();
  });
});


