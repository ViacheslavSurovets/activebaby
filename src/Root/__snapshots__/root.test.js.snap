// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BurgerMenuComponent Burger Menu Snapshot 1`] = `ShallowWrapper {}`;
